src/Arcus.Messaging.Abstractions/MessageHandling/MessageHandlerCollection.cs

```diff
@@ -20,6 +20,12 @@ public MessageHandlerCollection(IServiceCollection services)
             Guard.NotNull(services, nameof(services), "Requires a collection of services to register the message handling logic into");
             Services = services;
         }
+
+        /// <summary>
+        /// Gets or sets the unique message pump ID to identify the message pump throughout the application.
+        /// This ID can be used to get a reference of the previously registered message pump for which a message handler is registered.
+        /// </summary>
+        public string JobId { get; set; }
```

Comment:
I don't see why this property is called `JobId` but in the comments it is explained as to be the Message Pump ID.
(And then, I also do not see how this property (JobId / Pump ID) is set on a collection of message handlers ?

Reply:
If we set this ID when registering the message pump, extensions can make use of this pump. The message handlers are registered on the application services directly and may not use this, but a background job that wants to extend the message pump needs such an ID to know for which message pump it should run the background job. (Think about the background job that restarts message pumps on rotated credentials). If we pass this job ID, we don't need the consumer to pass it themselves.
This is also somewhat explained in the related issue here: https://github.com/arcus-azure/arcus.messaging/issues/333

Reply:
Should we then call it MessagePumpId, as you are identifying a message pump ?

Reply:
We already had a `JobId` as a property on the pump that was identifying it as such. The reason we choose 'job' is probably to identify any type of hosted service. Also, some kind of background job. We also have already a `Id` property on the message pump (not sure why, honestly), so if we would add another one, it would maybe make this more complicated, no?

Reply:
It's still not clear to me :)  Maybe I should use it to get a hold of it.
But if you say we already had a JobId as a property on the pump, then it means that  this  existing property is just promoted to be an interface member ?

Reply:
> But if you say we already had a JobId as a property on the pump, then it means that this existing property is just promoted to be an interface member ?

Yes, that's correct. 👍

Reply:
Ok, so it's not that easy to change the name of it, as it refers to an existing property.
I'm not entirely convinced regarding the name, and the comment that describes the property adds a bit to the confusion tbh. Maybe the comment can be improved ?

Reply:
Sure, we can do that. Any suggestions? As the first line of the comment is exactly the purpose of the property. Maybe it is the second line that adds confusion?

Reply:
The 2nd line is indeed confusing.
The 1st line is also somewhat confusing as the property is named 'JobId', while in the docs it is explained to be the Message Pump ID

Reply:
Changed it to:
```csharp
/// Gets or sets the unique ID to identify the job that runs a message pump during the lifetime of the application.
/// This ID can be used to get a reference of the previously registered message pump while registering message handlers and other functionality related to the message pump.
```